daemons/based/based_callbacks.c

```diff
@@ -1448,16 +1491,26 @@ cib_process_command(xmlNode *request, const cib_operation_t *operation,
     }
 
     if (!pcmk_is_set(call_options, cib_dryrun)
-        && pcmk__str_eq(section, XML_CIB_TAG_STATUS, pcmk__str_casei)) {
-        // Copying large CIBs accounts for a huge percentage of our CIB usage
+        && !pcmk__str_eq(op, PCMK__CIB_REQUEST_COMMIT_TRANSACT, pcmk__str_none)
+        && (pcmk_is_set(call_options, cib_transaction)
+            || pcmk__str_eq(section, XML_CIB_TAG_STATUS, pcmk__str_none))) {
+        /* Dry runs and commit-transaction requests must always make a copy.
+         *
+         * Requests in a transaction: we made a working CIB copy earlier in the
+         * call chain, and we need to accumulate changes in it.
+         *
+         * Status updates: copying large CIBs accounts for a huge percentage of
+         * our CIB usage, and this avoids some of it.
```

Comment:
It's not clear to me why it was safe to use `cib_zero_copy` with the status section, especially before the `cib_zero_copy` improvements in a previous commit of this PR.

Also, following those improvements, it might be possible to use `cib_zero_copy` with all replace operations and maybe others (e.g., erase), where we don't use the copy that we make... I'm not gonna check that right now.

Reply:
The status section is never written to disk, so no activation is needed

Reply:
Optionally writing to disk is part of activation, and yes, `cib_zero_copy` causes `new_cib == saved_cib` which currently breaks `activateCibXml()`. But that's easy to change in `activateCibXml()`, and `cib_zero_copy` does more than that. 

With `cib_zero_copy`, in `cib_perform_op()`, instead of copying the whole CIB, we operate on the live CIB. The upshot is that if we make some change to the CIB and then return an error, that erroneous change stays in the CIB. We don't recover, because we don't have a good copy to fall back to.

So what we're saying with the existing use of `cib_zero_copy` is "we don't care if the status section is flat-out wrong in memory." I'm wary of that.

Reply:
> Optionally writing to disk is part of activation, and yes, `cib_zero_copy` causes `new_cib == saved_cib` which currently breaks `activateCibXml()`. But that's easy to change in `activateCibXml()`, and `cib_zero_copy` does more than that.
> 
> With `cib_zero_copy`, in `cib_perform_op()`, instead of copying the whole CIB, we operate on the live CIB. The upshot is that if we make some change to the CIB and then return an error, that erroneous change stays in the CIB. We don't recover, because we don't have a good copy to fall back to.
> 
> So what we're saying with the existing use of `cib_zero_copy` is "we don't care if the status section is flat-out wrong in memory." I'm wary of that.

Only the configuration section is written to disk. The status section is always in-memory only. (The scheduler saves the entire CIB including status to its inputs directory, but the CIB itself never writes out status.)

Reply:
I don't think status changes can fail, because there's no schema to validate

Reply:
> Only the configuration section is written to disk. The status section is always in-memory only. (The scheduler saves the entire CIB including status to its inputs directory, but the CIB itself never writes out status.)

Yeah, I didn't intend to suggest otherwise. I meant that optionally writing [something] to disk is part of activation, and that `cib_zero_copy` controls other things too.

My concern is what happens if the status section is wrong *in memory*. We don't know if all four of the assumptions I mentioned above hold. (We can feel pretty good about the ACL one, as you explained.) If something requests status info from based, it could grab inaccurate info from an earlier erroneous update with `cib_zero_copy`.

I'm wary of dropping this now, because based on the comment it helped a lot with performance. However, it looks like there are some tricky and rare bugs lurking -- and potential future bugs, if new `cib_op_t` functions aren't atomic.